### PR TITLE
[READY TO MERGE] Port's RS's anti job camping to Anomoly

### PR DESCRIPTION
**Pull request by tacoguy7765093**
## About The Pull Request
Makes it so that if you played anomoly last shift, you will have to wait 10 minutes the next shift to be able to play again to give other people a chance to play shadekin. This should help with the same people camping the same job slot multiple shifts in the row and gives people that aren't as online a chance to play.

Code is by VerySoft and currently this affects anomaly only, can be put for future jobs.
## Changelog
:cl:
add: Added anti-job-camping to Anomalys
/:cl:

### CODE REVIEW

File: code/controllers/subsystems/job.dm [hunk 1]
```diff
@@ -141,3 +149,25 @@ SUBSYSTEM_DEF(job)
...
+		restricted_keys = json_decode(file2text(savepath))
+		fdel(savepath)
+
+/datum/controller/subsystem/job/Shutdown(Addr, Natural)
+	. = ..()
+	if(fexists(savepath))
+		fdel(savepath)
+	var/json_to_file = json_encode(shift_keys)
+	if(!json_to_file)
+		to_world("failed to write to json")
```
Comment (Kashargul) [inline code/controllers/subsystems/job.dm:165]: no to_world in live code

File: code/modules/mob/new_player/new_player_vr.dm [hunk 2]
```diff
@@ -74,6 +74,14 @@
 		if(points_left < 0 || traits_left < 0)
 			pass = FALSE
 			to_chat(src,"<span class='warning'>Your custom species is not playable. Reconfigure your traits on the VORE tab.</span>")
+//RS ADD START
+	if(J.camp_protection && round_duration_in_ds < CONFIG_GET(number/job_camp_time_limit))
+		if(SSjob.restricted_keys.len)
+			var/list/check = SSjob.restricted_keys[J.title]
+			if(client.ckey in check)
+				to_chat(src,"<span class = 'danger'>[J.title] is not presently selectable because you played as it last round. It will become available to you in [round((CONFIG_GET(number/job_camp_time_limit - round_duration_in_ds) / 600)] minutes, if slots remain open.</span>")
```
Comment (Kashargul) [inline code/modules/mob/new_player/new_player_vr.dm:82]: 1 opening bracket too much

**Review by Kashargul — CHANGES_REQUESTED** [review]

**Comment by tacoguy7765093** [comment]
Code has not been tested yet, will do so when I go home today, so I'll keep DNM for a couple of hours

File: code/game/machinery/computer/timeclock_vr.dm [hunk 3]
```diff
@@ -177,7 +177,17 @@
 		return
 	if(newassignment != newjob.title && !(newassignment in newjob.alt_titles))
 		return
+	//CHOMPadd START
+	if(newjob.camp_protection && round_duration_in_ds < CONFIG_GET(number/job_camp_time_limit))
+		if(SSjob.restricted_keys.len)
+			var/list/check = SSjob.restricted_keys[newjob.title]
+			if(usr.client.ckey in check)
+				to_chat(src,"<span class = 'danger'>[newjob.title] is not presently selectable because you played as it last round. It will become available to you in [round((CONFIG_GET(number/job_camp_time_limit) - round_duration_in_ds) / 600)] minutes, if slots remain open.</span>")
```
Comment (Kashargul) [inline code/game/machinery/computer/timeclock_vr.dm:185]: src can't work here.

File: code/modules/mob/new_player/new_player_vr.dm [hunk 4]
```diff
@@ -74,6 +74,14 @@
 		if(points_left < 0 || traits_left < 0)
 			pass = FALSE
 			to_chat(src,"<span class='warning'>Your custom species is not playable. Reconfigure your traits on the VORE tab.</span>")
+//RS ADD START
+	if(J.camp_protection && round_duration_in_ds < CONFIG_GET(number/job_camp_time_limit))
+		if(SSjob.restricted_keys.len)
+			var/list/check = SSjob.restricted_keys[J.title]
+			if(client.ckey in check)
+				to_chat(src,"<span class = 'danger'>[J.title] is not presently selectable because you played as it last round. It will become available to you in [round(CONFIG_GET(number/job_camp_time_limit - round_duration_in_ds) / 600)] minutes, if slots remain open.</span>")
```
Comment (Kashargul) [inline code/modules/mob/new_player/new_player_vr.dm:82]: src can't work here

**Review by Kashargul — CHANGES_REQUESTED** [review]

**Review by Kashargul — APPROVED** [review]

**Pull request merged**